source/PartialDirectedGraph.hpp

```diff
@@ -8,38 +8,48 @@
 
 namespace ADE {
 namespace Persistence {
+template <typename Type>
+class Table{
+public:
+  typedef Type data_type;
+  int version;
+  int table_type;  // 1: modificacion de valor, 2: modificacion de puntero
+  Node<Type> *ptr;
+  data_type data;
+  Table(int type, int ver) {
+  	ptr = nullptr;
+  	data = 100;
+  	table_type = type;
+  	version = ver;
+  }
+};
 
 template <typename Type>
 class PartialNode : public Node<Type> {
  public:
   typedef Type data_type;
-  PartialNode(data_type data, std::size_t const& out_ptrs_size,
-              std::size_t const& in_ptrs_size)
-      : Node<Type>(data, out_ptrs_size) {}
-
-  data_type get_data() { return *PointerMachine::Node<Type>::data_; }
-
-  data_type get_data(unsigned int version) { return *Node<Type>::data_; }
-
-  bool set_data(data_type const data) {
-    *Node<Type>::data_ = data;
-    return true;
-  }
-
-  bool set_ptr(PartialNode* ptr, unsigned int id) {
-    Node<Type>::forward_ = ptr;
-    return true;
-  }
-
-  PartialNode& operator[](std::size_t id) const {
-    return *dynamic_cast<PartialNode*>(&(Node<Type>::operator[](id)));
-  }
+  PartialNode(data_type data, std::size_t const& out_ptrs_size)
+      : Node<Type>(data, out_ptrs_size) {table = nullptr;}
 
   PartialNode& operator[](
       std::pair<std::size_t, unsigned int> id_version) const {
-    return *dynamic_cast<PartialNode*>(
-        &(Node<Type>::operator[](id_version.first)));
+    Node<Type> *next_node_ptr = nullptr;
+    if(table){
+    	if (table->version == id_version.second){
+    		next_node_ptr = table->ptr;
+    		return *dynamic_cast<PartialNode*>(next_node_ptr);
+    	}
+    }
+    	if (version == id_version.second){
+  	    	return *dynamic_cast<PartialNode*>(&(Node<Type>::operator[](id_version.first)));
+    	}
   }
+  Table<Type> *table;
+  void create_table(int type, int ver){
+  	table = new Table<Type>(type, ver);
```

Comment:
Solo almacenas un cambio, la tabla debe ser capaz de tener 2*p cambios
```suggestion
  	table = new Table<Type>[2 * p];
```